Dev

heupotato opened the pull request:
Pull branch dev into master

xuanphuc97 committed:
first commit algo

Add empty project BestTravel to algo branch

xuanphuc97 committed:
commit Algo.java Algorithm: Viết Code thuật toán

Viết code thuật toán để giải quyết bài toán và cho ra kết quả.
Kiểm thử thuật toán cho ra kết quả đều đúng và chưa tìm thấy ngoại lệ.

heupotato committed:
first commit gui

commit project file of best_travel project to gui branch

heupotato committed:
Phác thảo giao diện ứng dụng

Phác thảo giao diện của 2 main frame của ứng dụng: Main Menu và Start

heupotato committed:
UI Cửa sổ Main Menu

Cửa sổ bao gồm giao diện Menu, có các nút Start (để chạy chương trình) , History (Hiển thị lịch sử các lần sử dụng ứng dụng trước), Exit (Thoát ứng dụng)
Chứa 1 label miêu tả bài toán
Có thể tắt ứng dụng
Có mục để hiển thị thông tin nhóm phát triển ứng dụng
Logo/Img của ứng dụng
Phác thảo giao diện của cửa sổ Main Menu: ảnh đính kèm (interface.jpg)

heupotato committed:
Event Main Menu

Code các chức năng của các nút ở Main Menu:
- Nút Start: Mở cửa sổ Start để giải quyết bài toán đặt ra cho ứng dụng
- Nút History: Xem lịch sử sử dụng ứng dụng
- Nút Exit: Thoát ứng dụng
Code cửa sổ About us

heupotato committed:
Create UIStart.java

Có giao diện để dữ liệu nhập đầu vào, chạy chương trình và hiển thị kết quả lên màn hình. Trong đó giao diện đầu vào có thể nhập từ bàn phím hoặc bằng file
Có mục để quay lại cửa sổ Main Menu, thoát ứng dụng hoặc xem lịch sử sử dụng ứng dụng.
Minh hoạ giao diện cửa sổ Start: ảnh đính kèm (interface.jpg)

heupotato committed:
Event cửa sổ Start

Code các event của các phần tử trong cửa sổ Start
Sau mỗi lần chạy sẽ lưu lại kết quả chạy vào database

heupotato committed:
Cửa sổ History

Giao diện của cửa sổ History: Lịch sử các lần sử dụng ứng dụng, bao gồm INPUT, OUTPUT và ngày sử dụng ứng dụng
Các nút để thao tác với bảng lịch sử ứng dụng: Delete, Delete All
Nút Back để quay lại Main Menu

heupotato committed:
Event cửa sổ History

 Khi double-clicked vào 1 lịch sử bất kì sẽ hiển thị cửa sổ Start để có thể edit INPUT nhập vào trước đó và sử dụng lại
Nút Delete: Xoá một bản ghi lịch sử sử dụng ứng dụng
Nút Delete All: Xoá toàn bộ lịch sử sử dụng ứng dụng
Nút Back: Quay trở về cửa sổ Main Menu

heupotato committed:
Update UIMainMenu.java

Sửa text mô tả bài toán
Thêm cửa sổ khi chọn thoát ứng dụng

heupotato committed:
Update MainMenuEvent.java

Đổi màu của nút khi chuột trỏ vào
Thêm cửa sổ pop up khi nhấn nút exit

heupotato committed:
commit lại jdbc driver

heupotato committed:
commit jdbc driver

heupotato committed:
commit jdbc driver

heupotato committed:
Create mssql-jdbc-8.2.2.jre8.jar

heupotato committed:
Create mssql-jdbc-8.2.2.jre8.jar

heupotato committed:
Frame About Us

Hiển thị thông tin của nhóm phát triển

heupotato committed:
Merge pull request #14 from xuanphuc97/algo

Algo

heupotato committed:
Merge pull request #15 from xuanphuc97/database

Database

heupotato committed:
Merge pull request #16 from xuanphuc97/gui

Gui

heupotato committed:
Update StartEvent.java

- Thay đổi thông báo hiển thị khi không tìm được tổng thoả mãn yêu cầu đề
- Thêm label reset để reset các trường trong cửa sổ
- Fix bug: Sửa lại event reset text trong text area dùng để nhập dữ liệu
- Sửa lại màu của button browse khi nhập vào
- Merge code algo, database để hoàn thiện chức năng tính toán của cửa sổ (label 'click to see result')

heupotato committed:
Update UIStart.java

Thêm label reset để reset các thuộc tính trong cửa sổ

heupotato committed:
Update HistoryEvent.java

Merge code của database vào ui history để hiển thị lịch sử sử dụng ứng dụng ra bảng và hoàn thiện các tính năng của button delete và delete all

xuanphuc97 committed:
Update Database

Just clean code

heupotato committed:
Update UIHistory.java

Merge code database vào ui history để hiển thị thông tin lên bảng

heupotato committed:
Merge branch 'dev' of https://github.com/xuanphuc97/best-travel into dev

heupotato committed:
Revert "Merge branch 'dev' of https://github.com/xuanphuc97/best-travel into dev"

This reverts commit f3590cdb43c6b2da69950f02eae64b991d16d541, reversing
changes made to 6a097f6b1c90a7c57cfe36c9cf660262e0b78e7f.

heupotato committed:
Revert "Revert "Merge branch 'dev' of https://github.com/xuanphuc97/best-travel into dev""

This reverts commit 0f368470a579466b0edaa016225ac0c201dc3731.

heupotato commented:
@DACTechVN  @hientudact  Nhóm em đã tạo pull request branch dev vào master rồi ạ. 
Nhờ các anh review code giúp em với ạ. 
Chúng em xin cảm ơn ạ.

hungleba-dtvn commented on src/best_travel/UIMainMenu.java:
```diff
@@ -79,18 +98,18 @@ public UIMainMenu()
 		 *  part 1
 		 */ 
-		String text1 = "John and Mary want to travel between a few towns A, B, C ... Mary has on a sheet of paper a list of distances between these towns. John is tired of driving and he says to Mary that he doesn't want to drive more than t miles and he will visit only k towns.";
+		String text1 = "      John and Mary want to travel between a few towns A, B, C ... Mary has on a sheet of paper a list of distances between these towns. John is tired of driving and he says to Mary that he doesn't want to drive more than t miles and he will visit only k towns.";
```
- Đổi lại tên biến text1, text2 thành tên có nghĩa. Ví dụ: textContentTopic  ....

hungleba-dtvn commented on src/best_travel/DataBase.java:
```diff
@@ -0,0 +1,96 @@
+		}
+	}
+
+	public static void addHistory(String time, String list, int k, int t, String ouput) throws SQLException {
```
- Không nên dùng các biến có tên chung chung như k, t, tmp ... Nên đặt tên có nghĩa.

hungleba-dtvn commented on src/best_travel/UIHistory.java:
```diff
@@ -106,21 +105,49 @@ public UIHistory() {
+	public static void edittable() {
+		tbHistory.getTableHeader().setFont(new Font("Arial", Font.BOLD, 18));
+		tbHistory.setFont(new Font("Arial", Font.PLAIN, 15));
+	//	tbHistory.getTableHeader().setPreferredSize(new Dimension(100, 30));
```
- Remove code dư thừa

hungleba-dtvn commented on src/best_travel/StartEvent.java:
```diff
@@ -51,32 +67,149 @@ public void mouseClicked(MouseEvent e) {
+		 */
 		if (Text == "click to see result")
 		{
 			/*
```
- Function này xử lý mouseClicked(). Tuy nhiên, trong nó mình lại xử lý nhiều vấn đề khác, như đọc file, get time, add database -> nên tách riêng chúng thành từng hàm nhỏ và gọi chúng ở trong mouseClicked(). Không nên xử lý hết trong hàm mouseClick() vì sẽ quá dài và khó maintenance sau này.

hungleba-dtvn commented on src/best_travel/StartEvent.java:
```diff
@@ -51,32 +67,149 @@ public void mouseClicked(MouseEvent e) {
-			 * Process and calculate the distance
+			 * Process and calculate the sum of distances
+			 */
+			String defaut1Text = "type something here..."; 
```
- Sửa lại tên biến cho có nghĩa, không nên dùng tên default

hungleba-dtvn commented on src/best_travel/Algo.java:
```diff
@@ -0,0 +1,72 @@
+		return true;
+	}
+	/*
+	 *  Test
```
- Trong source code để release sẽ không để những code dư thừa khi mình test. Nên remove. Nếu khi em bảo vệ cần thì có thể copy lại để test.

hungleba-dtvn reviewed (commented):
Anh có một số góp ý sau:

Một pull request thì nên có một tả:
- Nội dung của pull request là gì?
- Tại sao pull request này được tạo ra?
- Làm thế nào để kiểm tra được pull request hoạt động đúng.

Một số vấn đề về coding:
- Tên biến nên đặt tên có nghĩa, không dùng tên chung or tên mặc định.
- Một function dài vì có quá nhiều xử lý nhỏ bên trong thì nên chia nhỏ ra để dễ quản lý.
- Các function như đọc file, get thời gian hiện tại thì nên đưa riêng ra để dễ tái sử dụng.
- Các source code dùng để debug nên remove khi commit.

xuanphuc97 committed:
Update Algo.java clean code

- Đặt lại tên biến cho rõ nghĩa
- Xóa nhưng phần không cần thiết

xuanphuc97 committed:
update Database.java clean code

- Thay đổi tên biến sao cho rõ nghĩa
- Thêm một số comment giải thích

heupotato committed:
Update UIMainMenu.java

Chỉnh sửa các tên biến thành tên có nghĩa

heupotato committed:
Update UIHistory.java

Xoá các dòng code dư thừa

heupotato committed:
Update AboutUs.java

Xoá các thư viện không cần dùng đến

heupotato committed:
Update MainMenuEvent.java

Xoá các thư viện không cần dùng đến

heupotato committed:
Update UIHistory.java

Xoá các thư viện không cần thiết
Chỉnh sửa truy cập của hàm setdatatable

heupotato committed:
Update UIMainMenu.java

Xoá các thư viện không cần thiết

heupotato committed:
Update UIStart.java

Xoá các thư viện không cần thiết

heupotato committed:
Update StartEvent.java

Thay đổi tên các biến thành các tên biến có ý nghĩa
Xoá các code dư thừa
Thêm một số comment để làm rõ code
Tách riêng các hàm trong event mouse_clicked thành các hàm getcurrenttime (để lấy thời gian hiện tại), readfile (đọc file) và newhistory (đưa input và output vào database)
